exporters/otlp/src/otlp_http_client.cc

```diff
@@ -981,16 +981,19 @@ void OtlpHttpClient::addSession(HttpSessionData &&session_data) noexcept
     return;
   }
 
-  opentelemetry::ext::http::client::Session *key = session_data.session.get();
-  ResponseHandler *handle = static_cast<ResponseHandler *>(session_data.event_handle.get());
-
-  handle->Bind(this, *key);
+  std::shared_ptr<opentelemetry::ext::http::client::Session> session = session_data.session;
+  std::shared_ptr<opentelemetry::ext::http::client::EventHandler> handle =
+      session_data.event_handle;
+  {
+    std::lock_guard<std::recursive_mutex> guard{session_manager_lock_};
+    static_cast<ResponseHandler *>(handle.get())->Bind(this, *session);
 
-  HttpSessionData &store_session_data = running_sessions_[key];
-  store_session_data                  = std::move(session_data);
+    HttpSessionData &store_session_data = running_sessions_[session.get()];
```

Comment:
Just curious , why can't we just use `session_data.session.get()` instead of `session.get()` for indexing here?